### DISCUSSION TITLE
Gtihub Api Zipball

### DISCUSSION THREAD

**Discussion by Fadwadhouaia** [community/community#27440] [Apps, API and Webhooks]
I'm trying to download zip file of a branch using the github api zipball and tarball but i have error 403
I'm using nuxtjs and this is my code 
var myHeaders = new Headers();
  myHeaders.append('Authorization', 'Bearer 'my-token');
  myHeaders.append('Content-Type', 'application/zip');
  myHeaders.append('Accept', 'application/vnd.github+json');
  myHeaders.append('Access-Control-Allow-Methods', ' GET, PUT, POST, DELETE, HEAD, OPTIONS');

  fetch('https://api.github.com/repos/{owner}/{repo}/zipball/{branchName}', {
    method: 'GET',
    headers: myHeaders,
    redirect: 'follow',
  });
![296177326_362974695991995_702887247682947754_n](https://user-images.githubusercontent.com/47947924/181463236-9a31b209-ec74-4ea8-bb7a-c0519c03c84b.png)

**Comment by airtower-luna** [comment]
I just tried myself, and the `Access-Control-Allow-Origin` header in the response confirmed my suspicion: The server forbids cross-origin requests from whatever your domain is.

```
access-control-allow-origin: https://render.githubusercontent.com
```

That's why the browsers reports a "CORS error". See the [article on Cross-Origin Resource Sharing (CORS) on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS).
